src/Http/Controllers/CartItemController.php

```diff
@@ -29,6 +30,8 @@ public function destroy(OrderItem $item)
     {
         $item->delete();
 
+        event(new RemovedFromCart($item));
```

Comment:
What if the item is removed by updating the qty to zero?

Isn't it better to trigger this event on the `OrderItem` model `deleted` event?

Reply:
The `RemovedFromCart` event should probably only be triggered if the Order is still in the cart status.

Or maybe we can leave that to the listener to check (if it matters)

Reply:
> See HandlesCartItems::remove(), that wouldn't trigger this event.

Why not just emitting the event also from there?

> And HandlesCartItems::empty() does a mass deletion [$this->items()->delete()], which wouldn't trigger even a model-event.

I would say that clearing the cart should be considered as a special action and we should probably add a `CartCleared` / `CartEmptied` event for that, I don't think it's justified to clear the items individually (on `HandlesCartItems::empty()`) just to have the event triggered

Reply:
> Why not just emitting the event also from there?

DRY 👨‍🍳

Besides, it's more logical.

Are there any downsides to that approach?

> `CartEmptied` event 

Totally makes sense to have a separate event for that.